sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageInputStream.java

```diff
@@ -111,6 +111,34 @@ protected StorageInputStream(long rangeOffset, final Long rangeLength,
         this.reposition(rangeOffset);
     }
 
+    /**
+     * Initializes a new instance of the StorageInputStream class.
+     *
+     * @param rangeOffset The offset of the data to begin stream.
+     * @param rangeLength How much data the stream should return after blobRangeOffset.
+     * @param chunkSize Holds the stream read size.
+     * @param contentLength The length of the stream to be transferred.
```

Comment:
why do we have both contentLenght and rangeLength here ?

Reply:
The contentLength is used to help calculate the rangeLength